vultr/resource_vultr_iso_private.go

```diff
@@ -107,7 +109,57 @@ func resourceVultrIsoDelete(ctx context.Context, d *schema.ResourceData, meta in
 	log.Printf("[INFO] Deleting iso : %s", d.Id())
 
 	if err := client.ISO.Delete(ctx, d.Id()); err != nil {
-		return diag.Errorf("error destroying ISO %s : %v", d.Id(), err)
+		// decode the error
+		var attachedErr struct {
+			Error  string `json:"error"`
+			Status int    `json:"status"`
+		}
+
+		if unmarshalError := json.Unmarshal([]byte(err.Error()), &attachedErr); unmarshalError != nil {
+			return diag.Errorf("error deleting ISO %s: parsing error %v in deleting ISO %s : %v", d.Id(), err.Error(), unmarshalError)
+		}
+
+		if !strings.Contains(attachedErr.Error, "is still attached to") {
+			return diag.Errorf("error deleting ISO %s: delete ISO error not related to attachment: delete error %+v", d.Id(), attachedErr)
+		}
+
+		parts := strings.Split(attachedErr.Error, " ")
+		ip := parts[len(parts)-1]
+		if parsedIP := net.ParseIP(ip); parsedIP == nil {
+			return diag.Errorf("error deleting ISO %s : failed to parse IP to which ISO is attached: %s", d.Id(), ip)
+		}
+
+		// default is 100 instances
+		var options govultr.ListOptions
+		for {
+			instances, meta, err := client.Instance.List(ctx, &options)
```

Comment:
This `meta` should be named something different since it overrides the meta that is passed into this function. Which causes issues when it gets handed off on line 146.

Other than this everything looks good. I am running through some tests locally now

Reply:
I missed that, thanks. The use of `interface{}` kind of masked this issue.

Please see https://github.com/vultr/terraform-provider-vultr/pull/131/commits/b8f03f2f328fae9355b64a224f7dd242acbf43f0 .